lib/helper/archiveMozillaHelper.py

```diff
@@ -0,0 +1,134 @@
+                    if complete_dir_url not in result_dict:
+                        result_dict[complete_dir_url] = {"folder_name": folder_name,
+                                                         "folder_datetime": "-".join(folder_name.split("-")[:6])}
+                    else:
+                        logger.warning("Duplicate folder url exist!!! [%s]" % complete_dir_url)
+        return result_dict
+
+    @staticmethod
+    def get_backfill_folder_list(input_backfill_days, input_backfill_repo, input_app_name, input_channel_name, input_history_folder_list):
+        """
+        base on backfill day and channel filter the archive folder list
+        @param input_backfill_days:
+        @param input_backfill_repo:
+        @param input_app_name:
+        @param input_channel_name:
+        @param input_history_folder_list:
+        @return:
+         {
+           "https://archive.mozilla.org/pub/firefox/nightly/2017/09/2017-09-15-10-01-21-mozilla-central/": {"folder_name": "2017-09-15-10-01-21-mozilla-central/",
+                                                                                                            "folder_datetime": "2017-09-15-10-01-21"}
+         }
+        """
+
+        current_utc_time = datetime.datetime.utcnow()
+        shift_days = datetime.timedelta(days=input_backfill_days)
+        backfill_start_date = current_utc_time - shift_days
+        backfill_start_year = backfill_start_date.strftime("%Y")
+        backfill_start_month = backfill_start_date.strftime("%m")
+        backfill_start_day = backfill_start_date.strftime("%d")
+        backfill_start_date_without_time = datetime.datetime.strptime(backfill_start_year + "-" + backfill_start_month + "-" + backfill_start_day, "%Y-%m-%d")
+        current_year = current_utc_time.strftime("%Y")
+        current_month = current_utc_time.strftime("%m")
+        DEFAULT_QUERY_CHANNEL_ARCHIVE_URL = "%s/pub/%s/%s" % (ArchiveMozillaHelper.DEFAULT_ARCHIVE_URL, input_app_name, input_channel_name)
+        query_dir_url = "%s/%d/%02d/" % (DEFAULT_QUERY_CHANNEL_ARCHIVE_URL, int(backfill_start_year), int(backfill_start_month))
+        total_folder_list = ArchiveMozillaHelper.get_folder_name_list(query_dir_url)
+        if backfill_start_year != current_year or backfill_start_month != current_month:
+            query_dir_url = "%s/%d/%02d/" % (DEFAULT_QUERY_CHANNEL_ARCHIVE_URL, int(current_year), int(current_month))
+            total_folder_list.extend(ArchiveMozillaHelper.get_folder_name_list(query_dir_url))
+        for history_folder_name in input_history_folder_list:
+            if history_folder_name not in total_folder_list:
+                total_folder_list.append(history_folder_name)
+        return ArchiveMozillaHelper.filter_backfill_period_data(backfill_start_date_without_time, current_utc_time, input_backfill_repo, total_folder_list, DEFAULT_QUERY_CHANNEL_ARCHIVE_URL)
```

Comment:
The `filter_backfill_period_data()` will return the dict object, but the method name is `get...list()`.
Also need some comments to describe the `input_app_name`, `input_channel_name`, and `input_history_folder_list`, for example, `https://archive.mozilla.org/pub/<input_app_name>/<input_channel_name>`